src/features/lightspeed/vue/views/fileOperations.ts

```diff
@@ -0,0 +1,159 @@
+  } else if (collectionMatch.length !== 1) {
+    throw new Error(
+      `Too many directories found for collection ${collectionName}!`,
+    );
+  }
+  const roleBaseDirUri = Uri.file(
+    `${collectionMatch[0].path}/roles/${roleName}`,
+  );
+  return roleBaseDirUri;
+}
+
+export async function fileExists(uri: Uri): Promise<boolean> {
+  try {
+    await workspace.fs.stat(uri);
+  } catch (e) {
+    if (e instanceof FileSystemError && e.code === "FileNotFound") {
+      return false;
+    }
+  }
+  return true;
+}
+
+export class FileOperations {
+  public async openLogFile(fileUrl: string) {
+    const logFileUrl = vscode.Uri.file(expandPath(fileUrl)).fsPath;
+    const parsedUrl = vscode.Uri.parse(`vscode://file${logFileUrl}`);
+    this.openFileInEditor(parsedUrl.toString());
+  }
+
+  public openFileInEditor(fileUrl: string) {
+    const updatedUrl = expandPath(String(fileUrl));
+    vscode.commands.executeCommand("vscode.open", vscode.Uri.parse(updatedUrl));
+  }
+
+  public async openRoleFolderInWorkspace(folderUrl: string, roleName: string) {
+    const folderUri = Uri.parse(expandPath(folderUrl));
+
+    if (workspace.workspaceFolders?.length === 0) {
+      workspace.updateWorkspaceFolders(0, null, { uri: folderUri });
+    } else {
+      await commands.executeCommand("vscode.openFolder", folderUri, {
+        forceNewWindow: true,
+      });
+    }
+
+    const mainFileUrl = `${folderUrl}/roles/${roleName}/meta/main.yml`;
+    console.log(`[ansible-creator] main.yml file url: ${mainFileUrl}`);
+    const parsedUrl = Uri.parse(`vscode://file${mainFileUrl}`);
+    console.log(`[ansible-creator] Parsed main.yml file url: ${parsedUrl}`);
+    this.openFileInEditor(parsedUrl.toString());
+  }
+
+  public async openFolderInWorkspacePlugin(
+    folderUrl: string,
+    pluginName: string,
+    pluginType: string,
+  ) {
+    const folderUri = vscode.Uri.parse(expandPath(folderUrl));
+
+    if (vscode.workspace.workspaceFolders?.length === 0) {
+      vscode.workspace.updateWorkspaceFolders(0, null, { uri: folderUri });
+    } else {
+      await vscode.commands.executeCommand("vscode.openFolder", folderUri, {
+        forceNewWindow: true,
+      });
+    }
+
+    // open the plugin file in the editor
+    const pluginTypeDir =
+      pluginType.toLowerCase() === "module"
+        ? "modules"
+        : pluginType.toLowerCase();
+    const pluginFileUrl = `${folderUrl}/plugins/${pluginTypeDir}/${pluginName}.py`;
+    const parsedUrl = vscode.Uri.parse(`vscode://file${pluginFileUrl}`);
+    this.openFileInEditor(parsedUrl.toString());
+  }
+
+  public async openFolderInWorkspace(folderUrl: string) {
+    const folderUri = Uri.parse(expandPath(folderUrl));
+
+    if (workspace.workspaceFolders?.length === 0) {
+      workspace.updateWorkspaceFolders(0, null, { uri: folderUri });
+    } else {
+      await commands.executeCommand("vscode.openFolder", folderUri, {
+        forceNewWindow: true,
+      });
+    }
```

Comment:
This looks like it can be moved into a reusable function and used in the other FileOperations methods.